agent/command/github_generate_token.go

```diff
@@ -0,0 +1,105 @@
+package command
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/evergreen/agent/internal"
+	"github.com/evergreen-ci/evergreen/agent/internal/client"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/google/go-github/v52/github"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+type githubGenerateToken struct {
+	// Owner of the repository. If not provided, the owner of the project is used.
+	Owner string `mapstructure:"owner" plugin:"expand"`
+
+	// Repo name. If not provided, the repo of the project is used
+	Repo string `mapstructure:"repo" plugin:"expand"`
+
+	// ExpansionName is what the generated token will be saved as.
+	ExpansionName string `mapstructure:"expansion_name"`
+
+	// Permissions to grant the token. If not provided, set to nil to grant all permissions.
```

Comment:
Can you please add a note that it can never be no permissions because we won't allow that use case?